### PR TITLE
makes follow flag not default for logs tail, adds some tests

### PR DESCRIPTION
**Pull request by zacharym**
@paetling cc @thieman

### CODE REVIEW

**Comment by paetling** [comment]
update your tests.  taking a look

File: dusty/cli/logs.py [hunk 1]
```diff
@@ -15,4 +17,4 @@
 
 def main(argv):
     args = docopt(__doc__, argv)
-    return tail_container_logs(args['<service>'], args['--tail'])
+    return tail_container_logs(args['<service>'], args['-f'], args['--tail'])
```
Comment (paetling) [inline dusty/cli/logs.py]: so i think @thieman moved away from having just normal tail be the default.  I think it is fine to change back, but might want to consult him

File: dusty/commands/logs.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 from .utils import exec_docker
 
-def tail_container_logs(app_or_service_name, lines=None):
+def tail_container_logs(app_or_service_name, follow=False, lines=None):
     container = get_dusty_containers([app_or_service_name])[0]
-    if lines is None:
-        exec_docker('logs', '-f', container['Id'])
-    else:
-        exec_docker('logs', '-f', '--tail={}'.format(lines), container['Id'])
+    args = ('logs',)
```
Comment (paetling) [inline dusty/commands/logs.py]: why not just make it a list. instead of making it tuple like this

File: dusty/commands/logs.py [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
+    if follow:
+        args += ('-f',)
+    if lines:
+        args += ('--tail={}'.format(lines),)
+    args += (container['Id'],)
+    exec_docker(args)
```
Comment (paetling) [inline dusty/commands/logs.py]: also i think you will need *args

File: dusty/commands/utils.py [hunk 4]
```diff
@@ -8,8 +8,8 @@
 def _executable_path(executable_name):
     return subprocess.check_output(['which', executable_name]).strip()
 
-def exec_docker(*args):
+def exec_docker(args):
```
Comment (paetling) [inline dusty/commands/utils.py]: you probably do not want to change exec_docker

File: dusty/cli/logs.py [hunk 5]
```diff
@@ -2,12 +2,14 @@
...
 Usage:
-  logs [--tail=NUM] <service>
+  logs [-f] [--tail=NUM] <service>
 
 Options:
+  -f          follow log output
   --tail=NUM  show NUM lines from end of file
 
 """
+import sys
```
Comment (thieman) [inline dusty/cli/logs.py]: extraneous

File: dusty/commands/logs.py (hunk 2 above)
Comment (thieman) [inline dusty/commands/logs.py]: Agreed, much more idiomatic to use lists for this sort of things. Tuples are immutable. This `+=` assignment technically works but it's very strange.

File: dusty/commands/utils.py (hunk 4 above)
Comment (thieman) [inline dusty/commands/utils.py]: What was the motivation for this?

File: dusty/commands/logs.py (hunk 2 above)
Comment (zacharym) [inline dusty/commands/logs.py]: I didn't realize that passing some *args would cast it as a tuple for you. nice

**Comment by zacharym** [comment]
@paetling @thieman this is updated

**Comment by paetling** [comment]
:ghost: :cow: seems good to me

**Pull request merged**